### PR TITLE
RSDK-3009 - Unpack mlClassifier easier

### PR DESCRIPTION
**Pull request by kharijarrett**
Resolved this issue. checkIfClassifierWorks was failing because we only ever attempt to unpack tensors listed under the term "probability" (empirically found) or "output0" (given default if MD does not exist).  In this particular case, the metadata did exist and was labeled output under the term "logits."

Rather than unpack based on the word "logits" as well, I've added a third and final check (for classifiers only). If you can't find "probability" or the default "output0", unpack the first output the map contains. This makes me nervous for future cases where we have a metadata-included detector, but the names are not what we expect.

Also slipped in here is a quick solution to RSDK-3034. Needed to do some length checking.

### CODE REVIEW

File: services/vision/mlvision/ml_model_test.go [hunk 1]
```diff
@@ -92,7 +92,7 @@ func TestAddingIncorrectModelTypeToModel(t *testing.T) {
 	test.That(t, classifier, test.ShouldNotBeNil)
 
 	err = checkIfClassifierWorks(ctx, classifier)
-	test.That(t, err, test.ShouldNotBeNil)
+	test.That(t, err, test.ShouldBeNil)
```
Comment (kharijarrett) [inline services/vision/mlvision/ml_model_test.go:95]: I think this change makes sense. That first tensor that comes out of a detector _could_ totally be a bunch of probabilities (or logits). It's not, but it totally could be.

**Review by bhaney — CHANGES_REQUESTED** [review]
In the engineering plan, we have a possibility of doing "Expose parameters for tflite model service" as a project. This ticket be the reason to make it a high priority.

We shouldn't be adding manual exceptions to the source code, and then editing the source code whenever we encounter a new exception.

File: services/vision/mlvision/ml_model.go [hunk 2]
```diff
@@ -108,12 +108,25 @@ func registerMLModelVisionService(
...
+	var me interface{}
+	if name != "" {
+		me = inMap[name]
+	} else {
+		if len(inMap) < 1 {
+			return nil, errors.New("could not unpack nonexistent first tensor")
+		}
+		for _, v := range inMap {
+			me = v
+			break
```
Comment (bhaney) [inline services/vision/mlvision/ml_model.go:123]: Go purposefully randomizes the order that it iterates through a map every time you run it. It is not deterministic. This will not work how you want

Comment (bhaney) [reply services/vision/mlvision/ml_model.go:123]: https://stackoverflow.com/questions/55925822/why-are-iterations-over-maps-random

File: services/vision/mlvision/ml_model_test.go (hunk 1 above)
Comment (kharijarrett) [reply services/vision/mlvision/ml_model_test.go]: Yeaaaah I take this back.

File: services/vision/mlvision/ml_model.go (hunk 2 above)
Comment (kharijarrett) [reply services/vision/mlvision/ml_model.go:120]: Okok makes sense but I hate it. Changed to only do this if we literally have just the one thing.

**Comment by kharijarrett** [comment]
Closing with the expectation that we will _soon_ be developing a more robust approach to this issue.
